### PR TITLE
ESWE-151 and ESWE-156 Enable pagination for …

### PR DESCRIPTION
**Pull request by steverendell**
…/offender-activities/{offenderNo}/work-history, and Remove 'EDU' filtering from api

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderActivitiesResource.java [hunk 1]
```diff
@@ -15,38 +17,37 @@
...
+import static java.util.Optional.ofNullable;
+
 @RestController
 @Api(tags = {"offender-activities"})
 @Validated
 @RequestMapping("${api.base.path}/offender-activities")
 public class OffenderActivitiesResource {
 
-    private final OffenderActivitiesService activitesService;
+    private final OffenderActivitiesService activitiesService;
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderActivitiesResource.java:34]: A spare endpoint which is not required

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderActivitiesResource.java [hunk 2]
```diff
@@ -15,38 +17,37 @@
...
-    @GetMapping("/{offenderNo}/current-work")
-    public OffenderActivities getCurrentWorkActivities(@PathVariable("offenderNo") @ApiParam(value = "The offenderNo of the prisoner", required = true) final String offenderNo) {
-        return activitesService.getCurrentWorkActivities(offenderNo);
+    public OffenderActivitiesResource(final OffenderActivitiesService activitiesService) {
+        this.activitiesService = activitiesService;
     }
 
     @ApiResponses({
         @ApiResponse(code = 500, message = "Unrecoverable error occurred whilst processing request.", response = ErrorResponse.class, responseContainer = "List")})
     @ApiOperation(value = "The work activities that this offender has been allocated to", notes = "This includes suspended activities", nickname = "getRecentStartedWorkActivities")
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderActivitiesResource.java:41]: breaking change but hidden in UI for now

File: src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/repository/OffenderProgramProfileRepository.java [hunk 3]
```diff
@@ -12,12 +14,14 @@
 
     @Query(
         value = """
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/repository/OffenderProgramProfileRepository.java:16]: Add explicit joins

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderActivitiesResource.java [hunk 4]
```diff
@@ -15,38 +17,37 @@
...
-        return activitesService.getStartedWorkActivities(offenderNo, earliestEndDate);
+    @GetMapping("/{offenderNo}/activities-history")
+    public Page<OffenderActivitySummary> getRecentStartedActivities(@PathVariable("offenderNo") @ApiParam(value = "The offenderNo of the prisoner", required = true) final String offenderNo,
+                                                                        @RequestParam(value = "earliestEndDate") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) @ApiParam(value = "Only include activities that have not ended or have an end date after the given date", example = "1970-01-02") final LocalDate earliestEndDate,
+                                                                        @RequestParam(value = "page", required = false) @ApiParam(value = "Target page number, zero being the first page", defaultValue = "0") final Integer pageIndex,
+                                                                        @RequestParam(value = "size", required = false) @ApiParam(value = "The number of results per page", defaultValue = "20") final Integer pageSize) {
+        final var pageIndexValue = ofNullable(pageIndex).orElse(0);
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderActivitiesResource.java:47]: this seems a bit naff - would be better to add a `defaultValue` onto `RequestParam`?

**Review by petergphillips — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderActivitiesResource.java (hunk 4 above)
Comment (steverendell) [reply src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderActivitiesResource.java:47]: ah probably, cheers. tbh I just copied this from another endpoint without looking too closely 🤦 ! Will change in the next one

**Pull request merged**